### ISSUE TITLE
On the Interplay Between Fine-tuning and Composition in Transformers

### ISSUE THREAD

**Issue by ymgw55** [ymgw55/MLPapers#131]
# Information
### URL
https://aclanthology.org/2021.findings-acl.201/
### Authors
[Lang Yu](https://aclanthology.org/people/l/lang-yu/), [Allyson Ettinger](https://aclanthology.org/people/a/allyson-ettinger/)
### Published at
ACL 2021

### Code
https://github.com/yulang/fine-tuning-and-composition-in-transformers
# Reading Motivation
#126 で紹介されていた, PAWS は人工的に作られているので relative distance を見れば 70% くらい判別できるという論文.
# Summary
事前学習済みの Transformer 言語モデル (= BERT など) は様々な NLP タスクで目覚ましいパフォーマンスを発揮している. しかし近年の研究から分かるのは, そのようなモデルの phrase-level での表現は単語内容の影響が大きく反映され, 精巧な構文の phrase 情報が不足している. 単語内容以上の phrase の意味情報を捉えるために, 動的な分散表現の能力について, fine-tuning の影響を調べた. 特に単語の overlap の多い, 敵対的な言い換えの分類タスク (PAWS) と感情分類タスク (SST) でモデルを fine-tuning した. fine-tuning の後, 先行研究に従い, overlap する単語の割合をコントロールした設定で, phrase の表現を分析した. そこから分かったのは, 分散表現について, fine-tuning は多くの場合, 構文情報に役立たないという事であった. ただ, SST での学習は特定のモデルに対しては僅かで局所的な恩恵がみられた. さらに追加の分析の中で, PAWS-QQP 中に, 構文情報の恩恵が得られないことを説明するかもしれない,  誤解を生む手がかりを見つけた. さらに SST の学習による局所的な恩恵のもととなる潜在的な要因について議論する.
# Background
fine-tuning で用いるデータセット.

- PAWS-QQP 
![スクリーンショット 2022-10-01 191704](https://user-images.githubusercontent.com/39965001/193404602-676543a5-8c46-47d8-b3d4-df263338f011.png)

- SST
https://huggingface.co/datasets/sst より引用
```bash
{'label': 0.7222200036048889,
 'sentence': 'Yet the act is still charming here .',
 'tokens': 'Yet|the|act|is|still|charming|here|.',
 'tree': '15|13|13|10|9|9|11|12|10|11|12|14|14|15|0'}
```
こんな感じで positive の度合いに応じて node にスコアがついている. 上と数値が異なるが, charming が一番スコアが高い.
![スクリーンショット 2022-10-01 192331](https://user-images.githubusercontent.com/39965001/193405610-529606fe-27fa-42fc-9f01-77272748d66a.png)
論文中では 5 -class の分類をしたが,  これは次のようなラベルで学習したと思われる. charming が一番スコアが高い.
```bash
{'ptb_tree': '(3 (2 Yet) (3 (2 (2 the) (2 act)) (3 (4 (3 (2 is) (3 (2 still) (4 charming))) (2 here)) (2 .))))'}
```

# Experiment
性能評価に用いるデータセット.
- BiRD
2単語の phrase に焦点を当てたデータセット. (STSのように) スコアと類似度の相関で性能評価.
- PPDB
言い換え分類タスク. accuracy で性能評価.

実験設定は, まず fine-tuning に PAWS-QQP と SST を用いて, uncontrolled な通常の BiRD, PPDB で性能評価. その後, 単語の意味内容による性能向上を ablation するために, controll されたデータセットで性能評価を行い, 通常の場合と比較する. BiRD では, AB-BA (e.g. law school / school law) の様な full-overlap なデータセット, 言い換え識別では２文が丁度 50 % overlap しているデータセットを用いる. このような controll をすることで単語の意味内容では判断できないので (e.g. law school/school law なら両方 law, school を含むため単語だけでは類似度が測れない), phrase level の理解を期待する (人間は law scool/school law の違いが理解できる). 

BiRD
- uncontrolled
![スクリーンショット 2022-10-01 195915](https://user-images.githubusercontent.com/39965001/193406396-90ae4fc6-d126-499f-bd07-f52047463c9c.png)
- controlled
![スクリーンショット 2022-10-01 195936](https://user-images.githubusercontent.com/39965001/193406399-fd4e74c0-53ea-4710-af01-1555f7b00700.png)

PPDB
- uncontrolled
![スクリーンショット 2022-10-01 200017](https://user-images.githubusercontent.com/39965001/193406445-b179d575-724a-4c08-902f-863b8dcfa8ff.png)
- controlled
![スクリーンショット 2022-10-01 200046](https://user-images.githubusercontent.com/39965001/193406452-b43413ce-352c-4a66-98fa-15b0f9af84ab.png)

結果としては, uncontrolled なデータセットでの性能評価と比べ, controlled な設定では性能が大きく落ちる. さらに fine-tuning することで pretrained よりも性能が悪くなるものもある.  とくに PAWS-QQP でパフォーマンスが下がる.

この原因を調べるために, 2文で初めて swap される単語について, swap 後どれだけ離れるかを用いて以下のスコアを定義する. 
![スクリーンショット 2022-10-02 161807](https://user-images.githubusercontent.com/39965001/193442717-8f6a6fd2-0db8-4c96-834d-f6dfc5af0088.png)
table 1 の 1文目の場合, first swapping word は specific で dist_swap = 4.

neg_prediciton では, dist_relative が大きくなる傾向があり,この情報だけである程度判断できそう. 
![スクリーンショット 2022-10-02 162347](https://user-images.githubusercontent.com/39965001/193442931-9cac6c64-14a3-45e3-831a-f2f352716935.png)

このスコアを用いて, 線形回帰を行ったモデルを Linerar CLF として fine-tuning 時のモデルの精度と比較.
![スクリーンショット 2022-10-02 162103](https://user-images.githubusercontent.com/39965001/193442822-fc61e334-e7f8-4e08-947a-291bb5046d43.png)

Table2 から分かるように, Linear CLF で 71.34 % の accuracy を達成しているので, 文の構造を見なくても, swap される単語の距離を見ればある程度正解できることが分かる. このことによって fine-tuning 時に, 文の構造ではなく, もっと簡単な性質を学習し, 実際に uncontrolled なデータセットで性能評価すると, 対処できずパフォーマンスが下がると考えている. これは PAWS が自動生成されたデータセットであり, そのことによる artifact であると考えられる. 

SST については, FIgure 2 の BERT の CLS で layer 4 で一番良いスコア. BERT に絞ってさらに実験する.
 SST を文の長さに分けてfine-tuning すると以下. この場合でも CLS で layer 4 の peak がみられ, 特に length 2 のときに他の長さよりスコアが良く, これは BiRD が bigram, unigram からなるデータセットであることに一貫性がある. ただし, Full dataset でのスコアが一番良い事から, 異なる文の長さのデータも性能向上に寄与していると分かる.

![スクリーンショット 2022-10-02 163145](https://user-images.githubusercontent.com/39965001/193443203-fe7e16f6-6d4a-43d9-8f6e-73fdb652cb51.png)

fine-tuning 後の, layer ごとの embedding について, pre-trained の場合の embedding との類似度を比較する. PAWS-QQP より, SST の方が変化が大きく, PAWS-QQP の方が簡単なタスクであること示唆している. 
![スクリーンショット 2022-10-02 163300](https://user-images.githubusercontent.com/39965001/193443252-44ad65eb-0829-497b-a350-8248869651fb.png)

# Consideration
- SST で uncontrolled な設定で若干の性能向上がみられたので, よりクオリティの高い構造情報を持つデータセットなら, さらなるパフォーマンスの改善が期待できる.
- artifact を含まない, よいデータセットを作るのは難しい.
